src/ItemLinkManagement/Public/New-ItemLink.ps1

```diff
@@ -45,7 +46,7 @@ function New-ItemLink {
             "Creating item '$_path'" | Write-Verbose
             New-Item -Path $_path -ItemType $ItemType -Value $_value -Force:$Force
         }catch {
-            Write-Error -Exception $_.Exception -Message $_.Exception.Message -Category $_.CategoryInfo.Category -TargetObject $_.TargetObject
+            Write-Error -ErrorRecord $_
```

Comment:
@joeltimothyoh the only problem with this is that the user of this cmdlet cannot control whether the error should be terminating or not with -ErrorAction.

Reply:
@joeltimothyoh looking at it again, the user can actually control it with `-ErrorAction`, but the `catch` block inherits its `$erroractionpreference` from the passed `-ErrorAction`.  But the behavior is implicit instead of being explictly specified, so the previous `if ($errorActionPreference -eq 'Stop') {} else { Write-Error -ErrorRecord $_ }` is better.

Reply:
This isn't all I changed in its related push. Strange why there isn't a way to view diffs between each force-push. The main bulk of changes are in https://github.com/theohbrothers/ItemLinkManagement/pull/6#discussion_r680581368 @leojonathanoh

Reply:
@joeltimothyoh from the looks of it again no, the `catch{}` block, as i initially suspected, inherits the `$ErrorActionPreference` from its first occurrence in the function. Hence the errors will always be terminating, and the user cannot turn that off but has to use a `try catch`.

Reply:
Unlike in the cmdlet [`New-Item`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-item?) `-Path` only accepts a single string rather than an array of them. To keep things simple, thus not having to handle the `ErrorActionPreference` value specified by the user.

I've before written more complex functions which support arrays of input objects and where `-ErrorAction Stop` is used strategically:

* https://github.com/joeltimothyoh/Remove-iCloudPhotosShortcut/blob/master/src/Remove-iCloudPhotosShortcut/helpers/Remove-QuickAccessItem.ps1

Reply:
Resolved, see https://github.com/theohbrothers/ItemLinkManagement/pull/6#discussion_r680603390